src/historyManager.js

```diff
@@ -35,12 +35,7 @@ export default class HistoryManager {
     const index = jsonPatchElement.path.split('/')[1];
     const value = actionType === 'undo' ? state[`_${index}`][0] : jsonPatchElement.value;
 
-    if (index === '0') {
-      await blocks.insert(value.type, value.data, {}, index + 1, true);
-      await blocks.move(index + 1, index);
-    } else {
-      await blocks.insert(value.type, value.data, {}, index, true);
-    }
+    await blocks.insert(value.type, value.data, {}, parseInt(index, 10), true);
```

Comment:
This type of functions do not have a clear way to validate that they have been executed since their response is `void`, in this case, what we do is to call the function with the specified data and the data type indicated in the documentation. The `insert` function is especially sensitive with the `index`, so we're parsing the value to always make sure it's an integer.

https://editorjs.io/blocks/#insert